lib/active_snapshot/models/concerns/snapshots_concern.rb

```diff
@@ -9,32 +9,16 @@ module SnapshotsConcern
     end
 
     def create_snapshot!(identifier: nil, user: nil, metadata: nil)
-      snapshot = snapshots.create!({
-        identifier: identifier,
-        user_id: (user.id if user),
-        user_type: (user.class.name if user),
-        metadata: (metadata || {}),
-      })
-
-      new_entries = []
+      snapshot = Snapshot.build_snapshot(self, identifier: identifier, user: user, metadata: metadata)
+      new_entries = snapshot.snapshot_items.map(&:attributes)
```

Comment:
Can this line be moved down below alongside `new_entries = new_entries.map` ?

Reply:
this goes inline with the comment about reset, to be able to save the snapshot I need to reset the snapshot_items otherwise the snapshot is not valid and will not save. The alternative would be to call `save(validate: false)` to save the snapshot, but I think this is better.